src/components/configuration/Configuration.ts

```diff
@@ -430,46 +418,33 @@ export class Configuration {
      * It assigns either a path specified by a user or undefined, depending on if a user specified a path and the specified path exists.
      * This method is asynchronous because it checks if a path specified by a user exists
      */
-    private async updatePathToRlsExecutableSpecifiedByUser(): Promise<void> {
+    private async updateUserRlsPath(): Promise<void> {
         function getRlsExecutable(): string | undefined {
             const configuration = Configuration.getConfiguration();
-
             const rlsConfiguration = configuration['rls'];
-
             if (!rlsConfiguration) {
                 return undefined;
             }
-
-            const pathToRlsExecutable = rlsConfiguration.executable;
-
-            if (!pathToRlsExecutable) {
+            const rlsPath = rlsConfiguration.executable;
+            if (!rlsPath) {
                 return undefined;
             }
-
-            return expandTilde(pathToRlsExecutable);
+            return expandTilde(rlsPath);
         }
 
         const logger = this.logger.createChildLogger('updatePathToRlsSpecifiedByUser: ');
 
-        const pathToRlsExecutable: string | undefined = getRlsExecutable();
-
-        if (!pathToRlsExecutable) {
-            this.pathToRlsSpecifiedByUser = undefined;
-
+        const rlsPath: string | undefined = getRlsExecutable();
+        if (!rlsPath) {
+            this.userRlsPath = undefined;
             return;
         }
-
-        const doesPathToRlsExecutableExist: boolean = await FileSystem.doesFileOrDirectoryExists(pathToRlsExecutable);
-
-        if (!doesPathToRlsExecutableExist) {
-            logger.error(`The specified path does not exist. Path=${pathToRlsExecutable}`);
-
-            this.pathToRlsSpecifiedByUser = undefined;
-
+        if (await FileSystem.which(rlsPath) == null) {
```

Comment:
where "C:\Users\user\.cargo\bin\cargo.exe"
ERROR: Invalid pattern is specified in "path:pattern".